src/wordcount.js

```diff
@@ -229,9 +230,22 @@ export default class WordCount extends Plugin {
 	_calculateWordsAndCharacters() {
 		const txt = modelElementToPlainText( this.editor.model.document.getRoot() );
 
-		this.characters = txt.length;
+		this.characters = txt.replace( /\n/g, '' ).length;
 
-		this.words = ( txt.match( /[_a-zA-Z0-9À-ž]+/gu ) || [] ).length;
+		// Groups:
+		// {L} - Any kind of letter from any language.
+		// {N} - Any kind of numeric character in any script.
+		// {M} - A character intended to be combined with another character (e.g. accents, umlauts, enclosing boxes, etc.).
+		// {Pd} - Any kind of hyphen or dash.
+		// {Pc} - A punctuation character such as an underscore that connects words.
+		const wordsMatchRegExp = env.features.isRegExpUnicodePropertySupported ?
```

Comment:
This shouldn't be created inside a frequently executed callback. It will affect the performance by taking some time, but, more importantly, by increasing GC pressure.